include/libp2p/crypto/key_validator/key_validator_impl.hpp

```diff
@@ -8,13 +8,13 @@
 
 #include <memory>
 
-#include <libp2p/crypto/key_generator.hpp>
+#include <libp2p/crypto/crypto_provider.hpp>
 #include <libp2p/crypto/key_validator.hpp>
 
 namespace libp2p::crypto::validator {
   class KeyValidatorImpl : public KeyValidator {
    public:
-    explicit KeyValidatorImpl(std::shared_ptr<KeyGenerator> key_generator);
+    explicit KeyValidatorImpl(std::shared_ptr<CryptoProvider> key_generator);
```

Comment:
change variable names as well

Reply:
I mean not only this entry, everywhere where required.

Reply:
thanks! did not notice that